app/build.gradle

```diff
@@ -191,9 +197,41 @@ dependencies {
   implementation project(":domain")
   implementation project(":utility")
 }
+repositories {
+  mavenCentral()
+}
+
 // The GeneratedMessageLite implementations of protobufs are depending on protobuf-java
 // instead of protobuf-lite after Android Studio 3.5,
 // The below command stops that from happening: https://github.com/google/tink/issues/282
 configurations {
   all*.exclude module: 'protobuf-java'
 }
+
+
+tasks.withType(Test) {
+  jacoco.includeNoLocationClasses = true
+}
+task jacocoTestReport(type: JacocoReport, dependsOn: ['testDebugUnitTest', 'createDebugCoverageReport']) {
+
+  reports {
+    xml.enabled = true
+    html.enabled = true
+    csv.enabled false
+  }
+  def fileFilter = [
+      '**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/Manifest*.*', '**/*Test*.*', 'android/**/*.*', '**/*$[0-9].*'
```

Comment:
Why are these filters needed?

Reply:
Without this, the android studio won't give the expected result of code coverage